contracts/utils/RolesBase.sol

```diff
@@ -0,0 +1,282 @@
+     */
+    function _rolesKey(address account) internal view virtual returns (bytes32 key) {
+        return keccak256(abi.encodePacked(ROLES_PREFIX, account));
+    }
+
+    /**
+     * @notice Internal function to get the roles of an account.
+     * @param account The address to get the roles for
+     * @return accountRoles The roles of the account in uint256 format
+     */
+    function _getRoles(address account) internal view returns (uint256 accountRoles) {
+        bytes32 key = _rolesKey(account);
+        assembly {
+            accountRoles := sload(key)
+        }
+    }
+
+    /**
+     * @notice Internal function to set the roles of an account.
+     * @param account The address to set the roles for
+     * @param accountRoles The roles to set
+     */
+    function _setRoles(address account, uint256 accountRoles) private {
+        bytes32 key = _rolesKey(account);
+        assembly {
+            sstore(key, accountRoles)
+        }
+    }
+
+    /**
+     * @notice Internal function to get the key of the proposed roles mapping.
+     * @param fromAccount The address of the current role
+     * @param toAccount The address of the pending role
+     * @return key The key of the proposed roles mapping
+     */
+    function _proposalKey(address fromAccount, address toAccount) internal view virtual returns (bytes32 key) {
+        return keccak256(abi.encodePacked(PROPOSE_ROLES_PREFIX, fromAccount, toAccount));
+    }
+
+    /**
+     * @notice Internal function to get the proposed roles of an account.
+     * @param fromAccount The address of the current role
+     * @param toAccount The address of the pending role
+     * @return proposedRoles_ The proposed roles of the account in uint256 format
+     */
+    function _getProposedRoles(address fromAccount, address toAccount) internal view returns (uint256 proposedRoles_) {
+        bytes32 key = _proposalKey(fromAccount, toAccount);
+        assembly {
+            proposedRoles_ := sload(key)
+        }
+    }
+
+    /**
+     * @notice Internal function to set the proposed roles of an account.
+     * @param fromAccount The address of the current role
+     * @param toAccount The address of the pending role
+     * @param proposedRoles_ The proposed roles to set in uint256 format
+     */
+    function _setProposedRoles(
+        address fromAccount,
+        address toAccount,
+        uint256 proposedRoles_
+    ) private {
+        bytes32 key = _proposalKey(fromAccount, toAccount);
+        assembly {
+            sstore(key, proposedRoles_)
+        }
+    }
+
+    /**
+     * @notice Internal function to add roles to an account.
+     * @dev emits a RolesAdded event.
+     * @dev Called in the constructor to set the initial roles.
+     * @param account The address to add roles to
+     * @param roles The roles to add
+     */
+    function _addRoles(address account, uint8[] memory roles) internal {
+        uint256 accountRoles = _getRoles(account);
```

Comment:
This is a pain to work with, because it requires:
```
uint8[] memory roles = new uint8[](1);
roles[0] = DISTRIBUTOR;
```
Maybe we can also add a function called _addRoles(address, uint8) that makes this easier for adding a single role.

Reply:
Yup will add functions `_addRole` and `_removeRole`

Reply:
Yup will add `_addRole` and `_removeRole` functions for convenience.

Reply:
@Foivos here is how you can actually do it. Wrapping it into a tuple
```
_addRoles(account, new uint8[]((DISTRIBUTOR)));
```